### PR TITLE
[ISSUE #9023] fix corner case config dataId 'cipher-' that cannot create

### PR DESCRIPTION
**Pull request by HotSince91**
Please do not create a Pull Request without creating an issue first.

## What is the purpose of the change

fix corner case config dataId 'cipher-' that cannot create

## Brief changelog

1. skip enc process when data-id equals encryption data-id prefix
2. a safer way to parse cipher algorithem 

## Verifying this change

success to create config 'cipher-' on page

Follow this checklist to help us incorporate your contribution quickly and easily:

* [ ] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [ ] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [ ] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

**Review by onewe — DISMISSED** [review]

**Comment by onewe** [comment]
Please retry ci

**Comment by KomachiSion** [comment]
The coverage of `EncryptionHandler.java` is down, Can you add more UT for your changes?

**Comment by HotSince91** [comment]
> The coverage of `EncryptionHandler.java` is down, Can you add more UT for your changes?

ok

File: plugin/encryption/src/test/java/com/alibaba/nacos/plugin/encryption/handler/EncryptionHandlerTest.java [hunk 1]
```diff
@@ -29,51 +29,110 @@
...
     @Before
     public void setUp() {
-        EncryptionPluginManager.join(new EncryptionPluginService() {
+        mockEncryptionPluginService = new EncryptionPluginService() {
             @Override
             public String encrypt(String secretKey, String content) {
-                return content;
+                return secretKey + content;
             }
-
```
Comment (li-xiao-shuang) [inline plugin/encryption/src/test/java/com/alibaba/nacos/plugin/encryption/handler/EncryptionHandlerTest.java:40]: Pay attention to code format

File: plugin/encryption/src/test/java/com/alibaba/nacos/plugin/encryption/handler/EncryptionHandlerTest.java [hunk 2]
```diff
@@ -29,51 +29,110 @@
...
         Assert.assertNotNull(pair);
     }
+
+    @Test
+    public void testCornerCaseDataIdAlgoParse() {
+        String dataId = "cipher-";
+        Exception e = null;
+        try {
+            EncryptionHandler.encryptHandler(dataId, "content");
+        } catch (IndexOutOfBoundsException ex) {
```
Comment (li-xiao-shuang) [inline plugin/encryption/src/test/java/com/alibaba/nacos/plugin/encryption/handler/EncryptionHandlerTest.java:89]: No exceptions need to be caught

**Review by li-xiao-shuang — APPROVED** [review]

**Pull request merged**